frappe/core/doctype/log_settings/test_log_settings.py

```diff
@@ -1,8 +1,128 @@
 # -*- coding: utf-8 -*-
-# Copyright (c) 2020, Frappe Technologies and Contributors
+# Copyright (c) 2022, Frappe Technologies and Contributors
 # License: MIT. See LICENSE
-# import frappe
+import frappe
 import unittest
 
+current = frappe.utils.now_datetime()
+past = frappe.utils.add_to_date(current, days=-4)
 class TestLogSettings(unittest.TestCase):
-	pass
+	@classmethod
+	def setUpClass(cls):
+		fieldnames = ['clear_error_log_after', 'clear_activity_log_after', 'clear_email_queue_after']
+		for fieldname in fieldnames:
+			frappe.set_value("Log Settings", None, fieldname, 1)
+
+	@classmethod
+	def tearDownClass(cls):
+		if frappe.db.exists({"doctype": "Activity Log", "subject": "Test subject"}):
+			activity_logs = frappe.get_all("Activity Log", filters=dict(subject='Test subject'), pluck='name')
+			for log in activity_logs:
+				frappe.db.delete("Activity Log", log)
+
+		if frappe.db.exists({"doctype": "Email Queue", "expose_recipients": "test@receiver.com"}):
+			email_queues = frappe.get_all("Email Queue", filters=dict(expose_recipients='test@receiver.com'), pluck='name')
+			for queue in email_queues:
+				frappe.db.delete("Email Queue", queue)
+
+		if frappe.db.exists({"doctype": "Error Log", "method": "test_method"}):
+			error_logs = frappe.get_all("Error Log", filters=dict(method='test_method'), pluck='name')
+			for log in error_logs:
+				frappe.db.delete("Error Log", log)
+
+	def test_create_activity_logs(self):
```

Comment:
1. `test_create_activity_logs`, `test_create_error_logs` & `test_create_email_queue` don't exactly test for those document creations accurately.
2. `doc.db_set` is much cleaner than using a set_value btw
3. These tests don't belong in `TestLogSettings`. I don't think these tests contribute to anything either. They're more like setUp methods for `test_delete_logs`